src/test/rustdoc-ui/run-directory/with-run-directory.rs

```diff
@@ -0,0 +1,8 @@
+// compile-flags:--test --test-run-directory={{src-base}}/run-directory
```

Comment:
Isn't this the default directory? Instead of having two different directories, can you have three revisions of the test?
1. Without a directory set, making sure the file is found
2. With a directory set to the default, making sure the file is found
3. With a directory set to something unhelpful, like /tmp, making sure the file is not found.

You can see instructions for writing revisions at https://rustc-dev-guide.rust-lang.org/tests/adding.html#revisions.

Reply:
So, compiletest does not set an explicit cwd for the rustdoc invocation:

https://github.com/rust-lang/rust/blob/ba6803e6b4f397c6e13e58595b6f1f7ef7c4a00a/src/tools/compiletest/src/runtest.rs#L1589

I think that is one of the reasons the tests need to `normalize-stdout-test`, because rustdoc outputs the test filename relative to the cwd. (which might be a good idea to implement/force? what do you think?)

Also, bootstrap does not set an explicit cwd for compiletest it seems:

https://github.com/rust-lang/rust/blob/ba6803e6b4f397c6e13e58595b6f1f7ef7c4a00a/src/bootstrap/test.rs#L993

I think that was why the first build failed on CI, because CI runs with a completely different cwd than my local `x test` invocation. Either way, I don’t really know what you mean by "default directory".

Reply:
Replied on zulip: https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/Source-based.20coverage.20for.20doctest/near/223757921